tracex_project/extraction/views.py

```diff
@@ -105,105 +103,89 @@ def get_context_data(self, **kwargs):
+                    df=trace,
+                    activity_key=activity_key,
                 ),
-                "xes_html": utils.Conversion.create_html_from_xes(
-                    single_trace_df_filtered
-                ).getvalue(),
+                "trace_table": utils.Conversion.create_html_table_from_df(trace),
                 "all_dfg_img": utils.Conversion.create_dfg_from_df(
-                    all_traces_df_filtered
+                    df=event_log,
+                    activity_key=activity_key,
+                ),
+                "event_log_table": utils.Conversion.create_html_table_from_df(
+                    event_log
                 ),
-                "all_xes_html": utils.Conversion.create_html_from_xes(
-                    all_traces_df_filtered
-                ).getvalue(),
             }
         )
 
-        # 5 .Generate XES files
-        single_trace_xes = utils.Conversion.dataframe_to_xes(
-            single_trace_df_filtered, "single_trace.xes"
+        self.request.session["trace"] = trace.to_json()
+        self.request.session["event_log"] = event_log.to_json()
+
+        return context
+
+    @staticmethod
+    def build_trace_df(filter_dict):
+        """Build the trace df based on the filter settings."""
+        trace_df = Orchestrator.get_instance().get_data()
+        trace_df_filtered = utils.DataFrameUtilities.filter_dataframe(
+            trace_df, filter_dict
         )
-        all_traces_xes = utils.Conversion.dataframe_to_xes(
-            all_traces_df_filtered, "all_traces.xes"
+
+        return trace_df_filtered
+
+    @staticmethod
+    def build_event_log_df(filter_dict, trace):
+        """Build the event log dataframe based on the filter settings."""
+        orchestrator = Orchestrator.get_instance()
+        # get only those traces from the database, that belong to the same condition as the newly extracted trace
+        condition = Cohort.manager.get(
+            pk=orchestrator.get_db_objects_id("cohort")
+        ).condition
+        event_log_df = utils.DataFrameUtilities.get_events_df(
+            Q(cohort__condition=condition)
         )
 
-        # 6. Store XES in session for retrieval in DownloadXesView
-        self.request.session["single_trace_xes"] = str(single_trace_xes)
-        self.request.session["all_traces_xes"] = str(all_traces_xes)
+        # concatenate the dataframes if they are not empty
+        if not event_log_df.empty and not trace.empty:
+            event_log_df = pd.concat(
+                [event_log_df, trace],
+                ignore_index=True,
+                axis="rows",
+            )
+        # filter the event log dataframe if it is not empty
+        if not event_log_df.empty:
+            event_log_df = utils.DataFrameUtilities.filter_dataframe(
+                event_log_df, filter_dict
+            )
+        # if event log is still empty, use the trace instead
+        elif not trace.empty:
+            event_log_df = trace
```

Comment:
Chat-GPT suggests the following refactored code for the if statemens. Be aware that you have to change the line with the return statement accordingly. I could not suggest that in this comment.

```suggestion
        if not event_log_df.empty:
            if not trace.empty:
                event_log_df = pd.concat([event_log_df, trace], ignore_index=True, axis="rows")
        event_log_df = utils.DataFrameUtilities.filter_dataframe(event_log_df, filter_dict)
    elif not trace.empty:
        event_log_df = trace

    return event_log_df

```